api/src/controllers/auth.controller.ts

```diff
@@ -38,18 +40,45 @@ export class AuthController {
     private userService: UserService,
     private jwtService: JwtService,
     private authService: AuthorizationService,
+    @InjectRepository(Invite) private inviteRepo: Repository<Invite>,
+    @InjectRepository(ProjectUser) private projectUserRepo: Repository<ProjectUser>,
     @InjectRepository(ProjectClient) private projectClientRepo: Repository<ProjectClient>,
   ) {}
 
   @Post('signup')
   @HttpCode(HttpStatus.OK)
   async signup(@Body() payload: SignupRequest) {
     if (!config.signupsEnabled) {
-      throw new ForbiddenException('Signups are disabled');
+      let invitesCount = await this.inviteRepo.count({
+        where: { email: payload.email, status: InviteStatus.Sent },
+      });
+
+      // Early exit if the user has no invitation.
+      if (invitesCount == 0) {
+        throw new ForbiddenException('Signups are invitation based only.');
+      }
     }
 
     const user = await this.userService.create(payload.name, payload.email, payload.password);
 
+    // accept project invites
+
+    const invites = await this.inviteRepo.find({
+      where: { email: payload.email, status: InviteStatus.Sent },
+      relations: ['project'],
+    });
+    invites.forEach(invite => (invite.status = InviteStatus.Accepted));
+    await this.inviteRepo.save(invites);
```

Comment:
Since there's many things being saved simultaneously, it'd be good to wrap them in a transaction in case something goes wrong in the middle of it. For reference see https://github.com/traduora/traduora/blob/2882ecc9c9a8bfae5b7279448ffa261b7e0e0b4c/api/src/controllers/term.controller.ts#L46

Reply:
Is there a way to make transations in multiple repositories at the same time?

Reply:
yes, typeorm can figure out the repos from the entities when using a transaction manager, see for example: https://github.com/traduora/traduora/blob/master/api/src/controllers/translation.controller.ts#L63

Reply:
thanks! done.